### PR TITLE
alot: fix ModuleNotFoundError of '_notmuch_config'

### PR DESCRIPTION
**Pull request by rien**
###### Description of changes

The `alot` package has been broken since #160575 because it relies on the `notmuch2` python binding (from the `notmuch` source) that tries to include a python module `_notmuch_config.py` generated by notmuch's configure script.

This PR fixes that issue by generating that file with nixpkgs. Aditionally, this makes the `notmuch2` package available as python module instead of buidling it solely for `alot`.

I can confirm `alot` is building again and working fine with `notmuch` 0.35.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by otavio — APPROVED** [review]

File: pkgs/applications/networking/mailreaders/alot/notmuch.nix [hunk 1]
```diff
@@ -14,6 +15,21 @@ buildPythonPackage {
 
   buildInputs = [ python notmuch cffi ];
 
+  # since notmuch 0.35, this package expects _notmuch_config.py that is normally
+  # generated by notmuch's configure script
+  postPatch = let
+    versionFile = writeTextFile {
```
Comment (roosemberth) [inline pkgs/applications/networking/mailreaders/alot/notmuch.nix:21]: I think it'd be better if we capture the file generated by the configure script as an additional output and consume that output here.

**Comment by teto** [comment]
I had had a look at this but couldn't finish the PR. A better fix might be to run the ./configure script in pkgs/development/python-modules/notmuch/default.nix  since that's what creates the file.

**Comment by rien** [comment]
@teto @roosemberth I agree that this could be regarded as kind of a "hack", I've added a new output `notmuch.bindingconfig` to notmuch which contains this generated file.

**Comment by rien** [comment]
I have also added `notmuch2` to python-modules, since it was not available as python package. This would be useful for developing `alot` locally.

I can remove this commit and make a new PR if desired.

**Pull request merged**